python/src/equistore/operations/slice.py

```diff
@@ -0,0 +1,395 @@
+        )
+    if samples_to_slice is not None:
+        assert isinstance(
+            samples_to_slice, Labels
+        ), "samples_to_slice must be passed as a equistore `Labels` object."
+        sample_names = (
+            tensormap.sample_names
+            if isinstance(tensormap, TensorMap)
+            else tensormap.samples.names
+        )
+        for name in samples_to_slice.names:
+            assert name in sample_names, (
+                "Invalid sample name '"
+                + name
+                + "' - doesn't exist in input `TensorMap`."
+            )
+    if properties_to_slice is not None:
+        assert isinstance(
+            properties_to_slice, Labels
+        ), "properties_to_slice must be passed as a equistore `Labels` object."
+        property_names = (
+            tensormap.property_names
+            if isinstance(tensormap, TensorMap)
+            else tensormap.properties.names
+        )
+        for name in properties_to_slice.names:
+            assert name in property_names, (
+                "Invalid property name '"
+                + name
+                + "' - doesn't exist in input `TensorMap`."
+            )
+
+    # Slice TensorMap
+    sliced_tensormap = _slice_tensormap(
+        tensormap,
+        samples_to_slice=samples_to_slice,
+        properties_to_slice=properties_to_slice,
+    )
+
+    # Calculate which blocks have been sliced to now be empty. True if any
+    # dimension of block.values is 0, False otherwise.
+    empty_blocks = np.array(
+        [np.any(np.array(block.values.shape) == 0) for _, block in sliced_tensormap]
+    )
+
+    # Issue warnings if some or all of the blocks are now empty.
+    if np.any(empty_blocks):
+        if np.all(empty_blocks):
+            warnings.warn(
+                "\nAll TensorBlocks in the sliced TensorMap are now empty, "
+                + "based on your choice of samples and/or properties to slice by. "
+            )
+        else:
+            warnings.warn(
+                "\nSome TensorBlocks in the sliced TensorMap are now empty, "
+                + "based on your choice of samples and/or properties to slice by. "
+                + "The keys of the empty TensorBlocks are:\n"
+                + str(tensormap.keys[empty_blocks])
+            )
+
+    return sliced_tensormap
+
+
+def slice_block(
+    tensorblock: TensorBlock,
+    samples_to_slice=None,
+    properties_to_slice=None,
+) -> TensorBlock:
+    """
+    Slices an input :py:class:`TensorBlock` along the samples and/or properties
+    dimension(s). `samples_to_slice` and `properties_to_slice` are
+    :py:class:`Labels` objects that specify the samples/properties
+    (respectively) names and indices that should be sliced, i.e. kept in the
+    output :py:class:`TensorBlock`.
+
+    Note that either `samples_to_slice` or `properties_to_slice`, or both,
+    should be specified as input. User warnings will be issued if the block is
+    empty after slicing.
```

Comment:
```suggestion
    should be specified as input.
```

And similar changes to the other functions